pyanaconda/modules/payloads/source/source_base.py

```diff
@@ -134,3 +136,31 @@ def tear_down_with_tasks(self):
         """
         task = TearDownMountTask(self._mount_point)
         return [task]
+
+
+class RPMSourceInterface(ABC):
+    """Interface class which has to be implented by all sources used by DNF payload."""
+
+    @abstractmethod
+    def generate_repo_configuration(self):
+        """Generate RepoConfigurationData structure.
+
+        This structure will be used by DNF payload in the main process.
+
+        FIXME: This is a temporary solution. Will be removed after DNF payload logic is moved.
+        """
+        pass
+
+
+class RPMMountingSourceBase(MountingSourceBase, ABC):
```

Comment:
I would go the other way, rename `MountingSourceBase` to `Mountable` and remove its parents, so it could be used as a mixin. Then we can have a proper subclass of `PayloadSourceBase` called `RPMSource`. I think it makes more sense. The current solution is little confusing.